lib/bundler/resolver.rb

```diff
@@ -163,6 +157,21 @@ def __dependencies
           dependencies
         end
       end
+
+      def metadata_dependencies(spec, platform)
+        return [] unless spec
+        # Only allow endpoint specifications since they won't hit the network to
+        # fetch the full gemspec when calling required_ruby_version
+        return [] if !spec.is_a?(EndpointSpecification) && !spec.is_a?(Gem::Specification)
```

Comment:
seems like both guard clauses could be replaced by `return [] unless [EndpointSpecification, Gem::Specification].include?(spec.class)`, since that would also check for `NilClass`.